agent/taskresource/volume/dockervolume_efs.go

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package volume
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/agent/config"
+
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/cihub/seelog"
+)
+
+const (
+	// EFSVolumePluginDriverName is the driver name of the ECS volume plugin.
+	EFSVolumePluginDriverName = "amazon-ecs-volume-plugin"
+	// EFSLocalDriverName is the name of Docker's local volume driver.
+	EFSLocalDriverName = "local"
+
+	efsVolumePluginDriverType = "efs"
+	efsLocalDriverType        = "nfs"
+
+	// Enums used by acs's api-2.json model.
+	efsIAMAuthEnabled           = "ENABLED"
+	efsTransitEncryptionEnabled = "ENABLED"
+)
+
+// EFSVolumeConfig represents efs volume configuration.
+type EFSVolumeConfig struct {
+	AuthConfig            EFSAuthConfig `json:"authorizationConfig"`
+	FileSystemID          string        `json:"fileSystemId"`
+	RootDirectory         string        `json:"rootDirectory"`
+	TransitEncryption     string        `json:"transitEncryption"`
+	TransitEncryptionPort int64         `json:"transitEncryptionPort"`
+	// DockerVolumeName is internal docker name for this volume.
+	DockerVolumeName string `json:"dockerVolumeName"`
+}
+
+// EFSAuthConfig contains auth config for an efs volume.
+type EFSAuthConfig struct {
+	AccessPointId string `json:"accessPointId"`
+	Iam           string `json:"iam"`
+}
+
+// GetLocalDriverOptions returns the options for creating an EFS volume using Docker's local volume driver.
```

Comment:
can we call it GetDockerLocalDriverOptions?

Reply:
yeah, updated